newIDE/app/src/ObjectsList/index.js

```diff
@@ -194,10 +201,6 @@ const ObjectsList = React.forwardRef<Props, ObjectsListInterface>(
       },
     }));
 
-    const [
-      renamedObjectWithContext,
-      setRenamedObjectWithContext,
-    ] = React.useState<?ObjectWithContext>(null);
```

Comment:
Here, you replaced the state hooks with 2 props. So the component does not really know how stable those 2 props are.
To make sure `ObjectsList` always uses props that are up to date in its hooks, you must make sure that they are included in their dependencies.
In particular, the methods `editName`, `rename` and `renderObjectMenuTemplate` miss `setRenamedObjectWithContext` in their dependencies.
Note: Here, your code works well because the props do not change at each interaction within ObjectsList, but it might be the case in the future, in which case you would have seen errors.

Reply:
`renderObjectMenuTemplate` I think doesn't use `setRenamedObjectWithContext`, but instead uses `preferences.values.userShortcutMap,`

![image](https://user-images.githubusercontent.com/21989259/208915244-5fe9ddb7-6ef2-41b1-81ba-a540ac1bbb1b.png)

Do you prefer to add both dependencies in `renderObjectMenuTemplate` or just one of them?

Reply:
And just a doubt: My Visual Studio always shows lot of errors regarding classes not found, like these:

![image](https://user-images.githubusercontent.com/21989259/208915521-d531fd03-776b-42da-a47d-768c0b0f23a7.png)

Any clue on how to make these errors dissapear?

Reply:
In the meanwhile I pushed the fix with the new props in the hooks

Reply:
> And just a doubt: My Visual Studio always shows lot of errors regarding classes not found, like these:
> 
> ![image](https://user-images.githubusercontent.com/21989259/208915521-d531fd03-776b-42da-a47d-768c0b0f23a7.png)
> 
> Any clue on how to make these errors dissapear?

I checked the readme and installed "Flow language support", now is ok!

Reply:
Great you saw that the Flow plugin was needed! You're one step closer to an ideal development environment for GDevelop